thingsboard_gateway/connectors/mqtt/mqtt_connector.py

```diff
@@ -27,29 +27,72 @@
 class MqttConnector(Connector, Thread):
     def __init__(self, gateway, config, connector_type):
         super().__init__()
+
+        self.__gateway = gateway                 # Reference to TB Gateway
+        self._connector_type = connector_type    # Should be "mqtt"
+        self.config = config                     # mqtt.json contents
+
         self.__log = log
-        self.config = config
-        self._connector_type = connector_type
-        self.statistics = {'MessagesReceived': 0,
-                           'MessagesSent': 0}
-        self.__gateway = gateway
+        self.statistics = {'MessagesReceived': 0, 'MessagesSent': 0}
+        self.__subscribes_sent = {}
+
+        # Extract main sections from configuration ---------------------------------------------------------------------
         self.__broker = config.get('broker')
-        self.__mapping = config.get('mapping')
-        self.__server_side_rpc = config.get('serverSideRpc', [])
-        self.__service_config = {"connectRequests": [], "disconnectRequests": []}
-        self.__attribute_updates = config.get("attributeUpdates")
-        self.__get_service_config(config)
-        self.__sub_topics = {}
+
+        self.__mapping = [
+            x for x in config.get('mapping')
+            if x.get("topicFilter") and x.get("converter")
```

Comment:
Please add here and in the same places logging like:
"topicFilter in config {...} - not found!"
"converter configuration not found in config: {}"
etc.

Logging will help to find a problem with configuration.

Reply:
That makes absolutely sense. Please check the last commit and let me know  it you like that.